test/unit/ObjectMakrTests/processFunctions.js

```diff
@@ -0,0 +1,35 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("processes the correct number of functions", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            var inheritance = mocks.mockInheritance();
+
+            // Act
+            objectMaker.functions = {};
```

Comment:
Then you're relying on there being 7 functions in your tests.

Pass them in the constructor.